packages/amplify-cli/src/context-extensions.ts

```diff
@@ -316,8 +316,10 @@ function attachTemplate(context: Context) {
       const data = {
         props,
       };
-      const directory = opts.directory;
-      const pathToTemplate = `${directory}/${template}`;
+      // If a directory was supplied, append a directory seprator.
+      // Otherwise, the template path will be use as-is.
+      const directory = opts.directory ? opts.directory + '/' : '';
+      const pathToTemplate = `${directory}${template}`;
```

Comment:
Added a comment to your other PR before realizing this was separate. Anyway, would be better to use [`path.join()`](https://nodejs.org/api/path.html#path_path_join_paths) here

Reply:
I've amended the code to conditionally use path.join if a directory is supplied in opts.dir.